nilearn/plotting/html_stat_map.py

```diff
@@ -1,148 +1,449 @@
+            output_json.write(json.dumps(params))
+
+    # save the colormap
+    if output_cmap is not None:
+        data = np.arange(0, n_colors)/(n_colors-1)
+        data = data.reshape([1, n_colors])
+        imsave(output_cmap, data, cmap=cmap, format=format)
+
+    return sprite
+
+
+def _custom_cmap(cmap, vmin, vmax, threshold=None):
+    # For internal use only
+
+    # Generation of a custom color map, with under-threshold "greyed" values
+    # This is taken from nilearn/plotting/displays.py#L754-L772
+
+    our_cmap = mpl_cm.get_cmap(cmap)
+    norm = colors.Normalize(vmin=vmin, vmax=vmax)
+    # edge case where the data has a single value
+    # yields a cryptic matplotlib error message
+    # when trying to plot the color bar
+    nb_ticks = 5 if norm.vmin != norm.vmax else 1
+    ticks = np.linspace(norm.vmin, norm.vmax, nb_ticks)
+    bounds = np.linspace(norm.vmin, norm.vmax, our_cmap.N)
+
+    # some colormap hacking
+    if threshold is None:
+        offset = 0
+    else:
+        offset = threshold
+    if offset > norm.vmax:
+        offset = norm.vmax
+    cmaplist = [our_cmap(i) for i in range(our_cmap.N)]
+    istart = int(norm(-offset, clip=True) * (our_cmap.N - 1))
+    istop = int(norm(offset, clip=True) * (our_cmap.N - 1))
+    for i in range(istart, istop):
+        cmaplist[i] = (0.5, 0.5, 0.5, 1.)  # just an average gray color
+    if norm.vmin == norm.vmax:  # len(np.unique(data)) == 1 ?
+        return
+    else:
+        our_cmap = colors.LinearSegmentedColormap.from_list(
+            'Custom cmap', cmaplist, our_cmap.N)
+        return our_cmap
 
 
 class StatMapView(HTMLDocument):
     pass
 
 
-def _to_papaya(cmap, norm, abs_min, vmax, symmetric_cmap):
-    """Create colormaps ('lookup tables') given to papaya viewer"""
-    n_colors = 100
-    if not symmetric_cmap:
-        x = np.linspace(norm(abs_min), norm(vmax), n_colors)
-        rgb = cmap(x)[:, :3]
-        return {'positive_cmap': _rgb_to_papaya(x, rgb),
-                'negative_cmap': None}
-    papaya = {}
-    x = np.linspace(norm(-vmax), norm(-abs_min), n_colors)
-    y = np.linspace(0, 1, n_colors)
-    rgb = cmap(x)[::-1, :3]
-    papaya['negative_cmap'] = _rgb_to_papaya(y, rgb)
-    x = np.linspace(norm(abs_min), norm(vmax), n_colors)
-    rgb = cmap(x)[:, :3]
-    papaya['positive_cmap'] = _rgb_to_papaya(y, rgb)
-    return papaya
-
-
-def _rgb_to_papaya(x, cmap):
-    """Transform (values, rgb array) to format expected by papaya"""
-    cmap = [[float(m)] + list(map(float, col)) for m, col in zip(x, cmap)]
-    return cmap
-
-
-def view_stat_map(stat_map_img, threshold=None, bg_img='MNI152',
-                  vmax=None, cmap='cold_hot', symmetric_cmap=True):
+def view_stat_map(stat_map_img, bg_img='MNI152', cut_coords=None,
+                  colorbar=True, title=None, threshold=None, annotate=True,
+                  draw_cross=True, black_bg='auto', cmap=cm.cold_hot,
+                  symmetric_cbar='auto', dim='auto', vmax=None,
```

Comment:
please keep the symmetric_cmap option to enable plotting anatomical images

Reply:
This has been addressed, hasn't it?